README.md

```diff
@@ -4,3 +4,37 @@
 # yeLLowhaMmer
 
 </div>
+
+A continuation of the [LLM hackathon](https://www.eventbrite.com/e/llm-hackathon-for-applications-in-materials-and-chemistry-tickets-868303598437) project, [*yeLLowhaMMer: A Multi-modal Tool-calling Agent for Accelerated Research Data Management*](https://github.com/bocarsly-group/llm-hackathon-2024).
+
+This repository will explore using the Jupyter AI plugin to provide an agentic interface to [*datalab*](https://github.com/datalab-org/datalab), with the idea of having this as an additional UI that *datalab* users can use to interact with their data, either deployed for their instance, or run locally.
+
+## Initial development tasks
+
+- [ ] Reproduce the hackathon project with the Jupyter AI plugin, probably by extending the `%%ai` cell magic to something like `%%yellowhammer` that includes the yellowhammer system prompt, and guides through the registration of a *datalab* API key and any provided LLM API keys.
+- [ ] Use yellowhammer to generate API examples for the underlying [*datalab-python-api*](https://github.com/datalab-org/datalab-python-api) package.
+- [ ] Consider deploying this as a JupyterHub that *datalab* instances can link to directly.
+- [ ] Integrate the results much more closely into *datalab* itself, i.e., attaching notebooks to the relevant samples, and recording the provenance of AI generated data recording.
+
+## Installation
+
+This repository uses [`uv`](https://docs.astral.sh/uv/) for the entire packaging workflow.
+Once you have installed `uv` following their documentation, you can install this repository by cloning and running `uv sync` in the root directory (optionally with `--dev` if you plan to develop it further).
+
+```shell
+git clone git@github.com:datalab-org/yellowhammer
+cd yellowhammer
+uv sync --dev
+```
+
+### Launching example notebooks
+
+You can launch the example notebook locally with `uv` too:
+
+```shell
+uv run jupyter lab examples/
```

Comment:
This works for me, but `uv run jupyter notebook examples/` doesn't load the environment correctly... any idea why?

Reply:
`notebook` isn't installed by default, perhaps its pulling from somewhere else? I can add it to the lockfile